### ISSUE TITLE
Calling someEndpoint.get() cancels any ongoing request

### ISSUE THREAD

**Issue by x8lucas8x** [kayak/redux-superapi#3]
Calling someEndpoint.get() will cancel any ongoing request, which is terrible when you have multiple identical container components in the same page. Ideally a request that have the same endpoint/arguments should just be ignored since the resolution of the initial request will eventually update all the components. I haven't tried to reproduce this behaviour for other http methods, but I assume they have the same issue.

**Comment by remiremi** [comment]
I'm not sure why you would want to trigger the exact same request several times simultaneously. I'd suggest using either the syncing or sync flag, something like

```
if (!state.someEndpoint.syncing) {
    dispatch(someEndpoint.get());
}
```

So that you don't trigger a new identical request if one is already running.

**Comment by remiremi** [comment]
The problem with running several times the same request concurrently is that it can lead to unexpected behaviors.

For example Component A starts a request. Then imagine Component B starts the same request knowing that the data has changed for whatever reason. With some bad luck, the request made by Component A might complete after Component B, overriding the latest result with old stale data.

It might sound far fetched but when it happens it's very hard to debug and fix.

**Comment by mikeengland** [comment]
In this case, it is probably best to just check the loading flag in the Redux store before executing a request. If you are executing many of these get requests in a short space of time, it is likely you have many identical components on the same page returning the same data. Just checking to ensure that one is in progress should fix this issue. What do you think @x8lucas8x ?

**Comment by x8lucas8x** [comment]
@remiremi Maybe I didn't make my point clear, but I'm not asking for redux-superapi to support running identical requests concurrently. There is no benefit in that. Also the current mechanism for using a key when different parameters are being used is enough. I simply believe that calling a dispatch(someEndpoint.get()) shouldn't cancel an ongoing identical request (i.e. same parameters/endpoint). Because in the end the first request will be resolved and all components that require its payload will be updated. Of course I could handle that myself like you proposed by implementing some logic that utilises loading and sync values. And in fact I had resorted to this already, but that's exactly what I was trying to avoid in the future when I reported this "issue" :).
